core/src/main/scala/fs2/Chunk.scala

```diff
@@ -200,20 +288,105 @@ object Chunk {
       (0 until size).foldLeft(z)((z,i) => f(z, at(i)))
     def foldRight[B](z: B)(f: (Boolean,B) => B): B =
       ((size-1) to 0 by -1).foldLeft(z)((tl,hd) => f(at(hd), tl))
+
+    // here be dragons; modify with EXTREME care
+    override def map[B](f: Boolean => B): Chunk[B] = {
```

Comment:
Well I'll be…

Not only is that dumb and yes, you're right that we aren't having `Byte`/`Boolean` unboxed, but it seems that the return type specialization isn't working somehow!  Upon closer examination of the bytecode:

```
      38: invokevirtual #128                // Method values:()[Z
      41: iload_2
      42: baload
      43: invokestatic  #192                // Method scala/runtime/BoxesRunTime.boxToBoolean:(Z)Ljava/lang/Boolean;
      46: invokeinterface #197,  2          // InterfaceMethod scala/Function1.apply:(Ljava/lang/Object;)Ljava/lang/Object;
      51: invokestatic  #315                // Method scala/runtime/BoxesRunTime.unboxToLong:(Ljava/lang/Object;)J
      54: lastore
```

So that's a bit weird…